src/plugins/socketmon/socketmon.cpp

```diff
@@ -282,57 +275,130 @@ static event_response_t tcpe_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
     if ( VMI_FAILURE == vmi_read(vmi, &ctx, sizeof(addrinfo), &addrinfo, NULL) )
         goto done;
 
-    ctx.addr = addrinfo.local;
-    if ( VMI_FAILURE == vmi_read(vmi, &ctx, sizeof(local), &local, NULL) )
-        goto done;
-
-    ctx.addr = local.pdata;
-    if ( VMI_FAILURE == vmi_read_addr(vmi, &ctx, &p1) )
-        goto done;
-
-    lip = read_ip_string(vmi, ctx, p1, inetaf.addressfamily);
-    if (!lip) goto done;
-
     rip = read_ip_string(vmi, ctx, addrinfo.remote, inetaf.addressfamily);
     if (!rip) goto done;
 
     if (!drakvuf_get_process_data(drakvuf, tcpe.owner, &owner_proc_data))
         goto done;
 
-    print_tcpe(drakvuf, info, s, owner_proc_data, inetaf.addressfamily, tcpe.state, lip, tcpe.localport, rip, tcpe.remoteport);
+    if (inetaf.addressfamily == AF_INET)
+        print_tcpe(drakvuf, info, s, owner_proc_data, inetaf.addressfamily, "127.0.0.1", tcpe.localport, rip, tcpe.remoteport);
+    else
+        print_tcpe(drakvuf, info, s, owner_proc_data, inetaf.addressfamily, "::1", tcpe.localport, rip, tcpe.remoteport);
 
 done:
     g_free(const_cast<char*>(owner_proc_data.name));
-    g_free(lip);
     g_free(rip);
     drakvuf_release_vmi(drakvuf);
-
     return 0;
 }
 
 static event_response_t tcpe_x86_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 {
-    return tcpe_cb<tcp_endpoint_x86, inetaf_x86, addr_info_x86, local_address_x86>(drakvuf, info);
+    return tcpe_old_cb<tcp_endpoint_x86, inetaf_x86, addr_info_x86>(drakvuf, info);
 }
 
-static event_response_t tcpe_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+static event_response_t tcpe_win81_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 {
-    return tcpe_cb<tcp_endpoint_x64, inetaf_x64, addr_info_x64, local_address_x64>(drakvuf, info);
+    return tcpe_old_cb<tcp_endpoint_win81_x64, inetaf_win81_x64, addr_info_x64>(drakvuf, info);
 }
 
-static event_response_t tcpe_win81_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+static event_response_t tcpe_win10_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 {
-    return tcpe_cb<tcp_endpoint_win81_x64, inetaf_win81_x64, addr_info_x64, local_address_x64>(drakvuf, info);
+    return tcpe_old_cb<tcp_endpoint_win10_x64, inetaf_win10_x64, addr_info_x64>(drakvuf, info);
 }
 
-static event_response_t tcpe_win10_x64_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+// uint16_t family = *(uint16_t*)(*(uint64_t*)(rcx + off1) + off2);
+static uint16_t tcp_get_family(vmi_instance_t vmi, addr_t rcx, addr_t build)
+{
+    addr_t ptr = 0, off1 = 0, off2 = 0;
+    uint16_t family = 0;
+
+    if (build == win_7_sp1_ver)
+    {
+        off1 = 0x18;
+        off2 = 0x14;
+    }
+    else if (build == win_10_1803_ver)
+    {
+        off1 = 0x10;
+        off2 = 0x18;
```

Comment:
Defining the offsets like this is very fragile. Can't these offset be found in some debug pdb that we could convert to a json?

Reply:
unfortunately not..
But they are hardcoded to a specific windows version

Reply:
In that case have these be defined in an array in `private.h` instead